swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1484,16 +1475,37 @@ public void setSeperator(final String _separator) {
 	 */
 	public boolean updateOption(final Long _mapping, final String _option) {
 
+		// 2021-02-28: @errael patched this method to deal with inconsistent
+		// updating of combo editor. See https://github.com/bpangburn/swingset/issues/85
+		
 		boolean result = false;
 
 		try (Model.Remodel remodel = optionModel.getRemodel()) {
 			final int index = remodel.getMappings().indexOf(_mapping);
 			if (index >= 0) {
+				boolean isSelectedItem = Objects.equals(_mapping, getSelectedMapping());
 				remodel.setOption(index, _option);
 				result = true;
```

Comment:
@errael See updated comments and removal of TODO per code review.

Reply:
Cool